src/otx/core/model/entity/detection.py

```diff
@@ -0,0 +1,138 @@
+"""Class definition for detection model entity used in OTX."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Union
+
+from torchvision import tv_tensors
+
+from otx.core.data.entity.base import OTXBatchLossEntity
+from otx.core.data.entity.detection import DetBatchDataEntity, DetBatchPredEntity
+from otx.core.model.entity.base import OTXModel
+from otx.core.utils.config import convert_conf_to_mmconfig_dict
+
+if TYPE_CHECKING:
+    from mmdet.models.data_preprocessors import DetDataPreprocessor
+    from omegaconf import DictConfig
+    from torch import nn
+
+
+class OTXDetectionModel(OTXModel[DetBatchDataEntity, DetBatchPredEntity]):
+    """Base class for the detection models used in OTX."""
+
+
+# This is an example for MMDetection models
+# In this way, we can easily import some models developed from the MM community
+class MMDetCompatibleModel(OTXDetectionModel):
+    """Detection model compatible for MMDet.
+
+    It can consume MMDet model configuration translated into OTX configuration
+    (please see otx.tools.translate_mmrecipe) and create the OTX detection model
+    compatible for OTX pipelines.
+    """
+
+    def __init__(self, config: DictConfig) -> None:
+        self.config = config
+        super().__init__()
+
+    def _create_model(self) -> nn.Module:
+        # import mmdet.models as _
+        from mmdet.registry import MODELS
+        from mmengine.registry import MODELS as MMENGINE_MODELS
+
+        # RTMDet-Tiny has bug if we pass dictionary data_preprocessor to MODELS.build
+        # We should inject DetDataPreprocessor to MMENGINE MODELS explicitly.
+        det = MODELS.get("DetDataPreprocessor")
+        MMENGINE_MODELS.register_module(module=det)
+
+        try:
+            model = MODELS.build(convert_conf_to_mmconfig_dict(self.config, to="tuple"))
+        except AssertionError:
+            model = MODELS.build(convert_conf_to_mmconfig_dict(self.config, to="list"))
+
+        return model
+
+    def _customize_inputs(self, entity: DetBatchDataEntity) -> dict[str, Any]:
+        from mmdet.structures import DetDataSample
+        from mmengine.structures import InstanceData
+
+        mmdet_inputs: dict[str, Any] = {}
+
+        mmdet_inputs["inputs"] = entity.images  # B x C x H x W PyTorch tensor
+        mmdet_inputs["data_samples"] = [
```

Comment:
In this context, we are leveraging MMEngine's data structures DetDataSample and InstanceData. This leads me to question why we need to create our own OTXDataEntity. Couldn't we reuse MMEngine's InstanceData structure? From what I can see, it offers a wealth of functionality and is well-defined. What are the reasons or concerns preventing us from utilizing their data structure?

Reply:
It's about the core concept. See my above comment.